src/tree/v2/ResourceGroupsItemCache.ts

```diff
@@ -109,8 +118,17 @@ export class ResourceGroupsItemCache {
         return currentItem;
     }
 
-    updateItemChildren(item: ResourceGroupsItem, children: ResourceGroupsItem[]): void {
+    updateItemChildren(item: ResourceGroupsItem, children: ResourceGroupsItem[]): InternalResourceGroupsItem[] {
         this.itemToChildrenCache.set(item, children);
-        children.forEach(child => this.itemToParentCache.set(child, item));
+        // cache the parent on the item
+        return children.map(child => Object.assign(child, { parent: item }));
```

Comment:
We should just make it writable or add a method to do so, I think that's cleaner than this `Object.assign()`